test/test_quantized_tensor.py

```diff
@@ -62,13 +63,18 @@ def _calculate_dynamic_qparams(X, dtype, reduce_range=False):
 
     return [scale.astype(np.float32), int(nudged_zero_point)]
 
+def get_supported_device_types():
+    return ['cpu', 'cuda'] if torch.cuda.is_available() and not TEST_WITH_ROCM else ['cpu']
+
 class TestQuantizedTensor(TestCase):
-    def test_qtensor(self):
+    @given(device=st.sampled_from(get_supported_device_types()),
+           dtype=st.sampled_from([torch.qint8, torch.quint8, torch.qint32]))
```

Comment:
The use of Hypothesis here seems wrong. You don't want to randomly sample from the space of six options. You want to just exhaustively run all of these tests! Hypothesis is for randomized testing and it doesn't seem like randomized testing is buying you anything here.

Reply:
Guess, I should have read hypothesis docs carefully) Sorry, never worked with it before. Reworked everything to plain loops.